Homework_17-18_Networking/README.md

```diff
@@ -0,0 +1,188 @@
+
+```
+**Established only:**
+```
+[root@localhost superadmin]#  tcpdump -i eth0 dst 192.168.0.220 and port 22 and 'tcp[13] & 8 == 8'
+tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
+listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
+05:06:29.036213 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 2726740141:2726740182, ack 199548490, win 502, options [nop,nop,TS val 2774576705 ecr 2922395], length 41
+05:06:29.056891 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 41:1553, ack 22, win 502, options [nop,nop,TS val 2774576726 ecr 2922414], length 1512
+05:06:29.069420 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 1553:1601, ack 1302, win 501, options [nop,nop,TS val 2774576738 ecr 2922419], length 48
+05:06:29.088763 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 1601:1617, ack 1666, win 501, options [nop,nop,TS val 2774576758 ecr 2922441], length 16
+05:06:29.128898 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 1617:1661, ack 1666, win 501, options [nop,nop,TS val 2774576798 ecr 2922489], length 44
+05:06:29.129580 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 1661:1729, ack 1710, win 501, options [nop,nop,TS val 2774576799 ecr 2922489], length 68
+05:06:29.138009 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 1729:2101, ack 1794, win 501, options [nop,nop,TS val 2774576807 ecr 2922493], length 372
+05:06:29.141651 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 2101:2753, ack 2126, win 501, options [nop,nop,TS val 2774576811 ecr 2922500], length 652
+05:06:29.154641 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 2753:2865, ack 2154, win 501, options [nop,nop,TS val 2774576824 ecr 2922514], length 112
+05:06:29.489831 IP 192.168.3.86.52356 > localhost.localdomain.ssh: Flags [P.], seq 2865:3257, ack 2698, win 501, options [nop,nop,TS val 2774577159 ecr 2922849], length 392
+```
+3. Close session. Find in tcpdump output packets that are related to TCP session closure.
+```
+[root@localhost superadmin]#  tcpdump -i eth0 dst 192.168.0.220 and port 22 and 'tcp[13] & 1 == 1'
+tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
+listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
+05:17:25.145146 IP 192.168.3.86.52450 > localhost.localdomain.ssh: Flags [F.], seq 849559791, ack 435058857, win 501, options [nop,nop,TS val 2775232808 ecr 3578504], length 0
+```
+4. run tcpdump and request any http site in separate session. Find HTTP request and answer packets with ASCII data in it.  Tcpdump command must be as strict as possible to capture only needed packages for this http request.
+```
+[root@localhost superadmin]# tcpdump -A  'tcp port 80 and (((ip[2:2] - ((ip[0]&0xf)<<2)) - ((tcp[12]&0xf0)>>2)) != 0)'
```

Comment:
Will this command allow us to get data specific only to your request or it will get all traffic with tcp protocol, port 80, etc?

Reply:
It prints all IPv4 HTTP packets to and from port 80, i.e. print only packets that contain data, not, for example, SYN and FIN packets and ACK-only packets.
I took this command from EXAMPLES in man tcpdump.

Reply:
And what if we will someone will send a query to another destination while we are sniffing a specific one?

Reply:
ok, i can do like this:

Reply:
Yeah, that should be fine.
The main purpose of the task is to teach to specify command as strictly as possible.

Reply:
I focused on http and forgot about src/dst. Thanks a lot.

Reply:
FYI, you can use `host 192.168.3.140` instead of `dst 192.168.3.140 or src 192.168.3.140`